Tests/SwiftSyntaxMacroExpansionTest/LexicalContextTests.swift

```diff
@@ -0,0 +1,298 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+//==========================================================================//
+// IMPORTANT: The macros defined in this file are intended to test the      //
+// behavior of MacroSystem. Many of them do not serve as good examples of   //
+// how macros should be written. In particular, they often lack error       //
+// handling because it is not needed in the few test cases in which these   //
+// macros are invoked.                                                      //
+//==========================================================================//
+
+import SwiftSyntax
+import SwiftSyntaxMacroExpansion
+import SwiftSyntaxMacros
+import SwiftSyntaxMacrosTestSupport
+import XCTest
+
+extension PatternBindingSyntax {
+  /// When the variable is declaring a single binding, produce the name of
+  /// that binding.
+  fileprivate var singleBindingName: String? {
+    if let identifierPattern = pattern.as(IdentifierPatternSyntax.self) {
+      return identifierPattern.identifier.trimmedDescription
+    }
+
+    return nil
+  }
+}
+
+private extension TokenSyntax {
+  var asIdentifierToken: TokenSyntax? {
+    switch tokenKind {
+    case .identifier, .dollarIdentifier: return self.trimmed
```

Comment:
Slightly more efficient

```suggestion
    case .identifier, .dollarIdentifier: return self.text
```